### PR TITLE
Refactor Vue Router and Authentication infrastructure

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR rewrites major chunks of the codebase to support using `oidc-client-ts` as our core authentication library instead of `keycloak-js`. It also contains a rewrite of the Vue Router layer so that it can properly support dynamic route entrypoints as a properly functioning SPA. Other numerous fixes have been included to help shore up the application, including a security audit and dependency update pass.

- 8b3af09 Routine npm dependency and RedHat image updates
- fd4c9ba Change configmap environment values to support OIDC instead of KC
- dd34c47 Fix remaining linting, testing and compilation issues
- 1511830 Implement componetized progress loader bar and integration to router
- 1fabe1e Add bcgov styling for header and navbar
- 8dfff8c Implement explicit oidc flow routes and views for soft redirection
- e08a009 Attempt to perform vue-router navigation after /oidc/callback
- 2f11a23 Change vue-router to depend on authStore instead of authService
- 0a5df11 Update components to use new authStore methods
- eaec877 Update base views and userStore to use new store getters
- c9dd25f Switch axios service interceptor to leverage new store getter functions
- a174e77 Drop generateGetters higher order function support
- 46699b5 Update authService to use a static userManager instance
- 6235b87 Remove suspense support from App.vue
- 4d2c896 Refactor main initializeApp to force a pinia configStore init
- d92961c Rename ObjectFileDetailsView to DetailObjectsView
- 8614698 Consolidate vue router to handle oidc redirection and logouts
- 3d7ebc7 Update authStore and configStore to support state masking
- b85f38f Implement higher order generateGetters function and supporting types
- 8ff6f68 Add npm debug script to frontend package
- 6fbb089 Add generic createProps transform function to router
- 6b6f6c4 Refactor core routing logic to utilize new authService
- ba98368 Refactor main startup code to use new blocking initialization flows
- 6eca2e5 Implement new AuthService singleton service
- bfbe322 Implement new ConfigService singleton service
- 9e9141a Add oidc-client-ts library dependency
- 471f278 Add dist folder to local ts-node-dev watchlist

Without these changes, we would not be able to proceed with public sharing and other general frontend application behavior.

[SHOWCASE-3002](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3002)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Things will break. You will need to realign your local.json for things to work again. Do `npm ci` as nearly all dependencies have been updated.

### CODE REVIEW

File: frontend/src/components/bucket/BucketsTable.vue [hunk 1]
```diff
@@ -78,12 +79,12 @@ const displayPermissionsIcon = (bucket: Bucket) => {
             v-if="data.bucketName.length > 150"
             v-tooltip.bottom="{ value: data.bucketName }"
           >
-            <router-link :to="{ name: RouteNames.ListObjects, query: { bucketId: data.bucketId } }">
+            <router-link :to="{ name: RouteNames.LISTOBJECTS, query: { bucketId: data.bucketId } }">
```
Comment (kamorel) [inline frontend/src/components/bucket/BucketsTable.vue:82]: Style: Do we want to underscore multi word consts for easier readability?

File: frontend/src/components/layout/ProgressLoader.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const { getLoadingMode, getLoadingValue } = storeToRefs(appStore);
+</script>
+
+<template>
+  <!-- eslint-disable vue/attribute-hyphenation -->
+  <ProgressBar
+    :mode="getLoadingMode"
+    :showValue="false"
+    :value="getLoadingValue"
+  />
```
Comment (kamorel) [inline frontend/src/components/layout/ProgressLoader.vue:17]: Is this disable necessary? Was it not correctly accessing the PrimeVue attribute when hyphenated?

**Review by kamorel — APPROVED** [review]

**Review by TimCsaky — APPROVED** [review]

File: frontend/src/components/layout/ProgressLoader.vue (hunk 2 above)
Comment (jujaga) [reply frontend/src/components/layout/ProgressLoader.vue:17]: You are right we can do it in hyphenated style and it still works.

File: frontend/src/App.vue [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
+
+import { AppLayout, Navbar, ProgressLoader } from '@/components/layout';
+import { useAppStore, useAuthStore, useUserStore } from '@/store';
+
+const appStore = useAppStore();
+const { getIsLoading } = storeToRefs(appStore);
+
+onBeforeMount(async () => {
+  appStore.beginDeterminateLoading();
+  await useAuthStore().init();
```
Comment (TimCsaky) [inline frontend/src/App.vue:17]: i thought useAuthStore().init(); was already called when you initializeApp() in main.ts

Comment (jujaga) [reply frontend/src/App.vue:17]: We might not necessarily need it here as in theory this should have either already been done in main.ts or done in children components; App.vue itself has no requirements to know about the authentication state; but the children components do.

**Pull request merged**